compiler/rustc_mir/src/transform/coverage/graph.rs

```diff
@@ -344,6 +492,175 @@ fn bcb_filtered_successors<'a, 'tcx>(
     .filter(move |&&successor| body[successor].terminator().kind != TerminatorKind::Unreachable)
 }
 
+/// Maintains separate worklists for each loop in the BasicCoverageBlock CFG, plus one for the
+/// CoverageGraph outside all loops. This supports traversing the BCB CFG in a way that
+/// ensures a loop is completely traversed before processing Blocks after the end of the loop.
+#[derive(Debug)]
+pub(crate) struct TraversalContext {
```

Comment:
Another piece of code I think would benefit from some unit tests

Reply:
I added a FIXME comment to add unit tests here as well.